app/controllers/campaigns_controller.rb

```diff
@@ -67,10 +67,10 @@ def users
   def alerts
     alerts = []
     @campaign.course_ids.each do |course_id|
-      alert = Alert.find_by(course_id: course_id)
-      alerts << alert.id unless alert.nil?
+      alert = Alert.where(course_id: course_id)
+      alert.each { |alert| alerts << alert.id unless alert.nil? }
     end
-    alerts
+    render json: { campaign_alerts: alerts }
```

Comment:
@ragesoss do you mean by scope adding it to the campaign model? That makes sense, yes. I was also thinking about adding a reducer and action and therefore creating here a json so we are able to call something like /campaigns/:campaign_slug/alerts.json and receive a json with the information. 
I am not sure what would be better specially in terms of performance or good practices. Any suggestion?